packages/sitecore-jss-react/src/enhancers/withComponentFactory.tsx

```diff
@@ -12,11 +13,22 @@ export interface ComponentFactoryProps {
 export function withComponentFactory<T extends ComponentFactoryProps>(
   Component: React.ComponentClass<T> | React.FC<T>
 ) {
-  return function WithComponentFactory(props: T) {
+  /**
+   * @param {T} props - props to pass to the wrapped component
+   * @returns {JSX.Element} - the rendered component
+   */
+  function WithComponentFactory(props: T) {
+    const context = useContext(ComponentFactoryReactContext);
     return (
       <ComponentFactoryReactContext.Consumer>
```

Comment:
Do you need to wrap it by `ComponentFactoryReactContext.Consumer` in this case? Seems like `useContext` is enough